nuxeo-coldstorage/src/test/java/org/nuxeo/coldstorage/helpers/S3TestColdStorage.java

```diff
@@ -0,0 +1,97 @@
+ * Contributors:
+ *     Abdoul BA<aba@nuxeo.com>
+ */
+
+package org.nuxeo.coldstorage.helpers;
+
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+
+import org.junit.Test;
+import org.nuxeo.coldstorage.S3ColdStorageFeature;
+import org.nuxeo.coldstorage.S3TestHelper;
+import org.nuxeo.ecm.core.api.*;
+import org.nuxeo.ecm.core.blob.ManagedBlob;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+
+@Features(S3ColdStorageFeature.class)
+@Deploy("org.nuxeo.coldstorage.test:OSGI-INF/test-s3-coldstorage-contrib.xml")
+public class S3TestColdStorage extends AbstractTestColdStorageHelper {
+
+    protected S3TestHelper s3TestHelper = S3TestHelper.getInstance();
+
+    @Override
+    protected String getBlobProviderName() {
+        return "glacier";
+    }
+
+
+    @Test
+    public void shouldBeingRetrieved() {
+        DocumentModel documentModel = createFileDocument(DEFAULT_DOC_NAME, true);
+
+        // move the blob to cold storage
+        documentModel = moveContentToColdStorage(session, documentModel.getRef());
+        session.saveDocument(documentModel);
+        // request a retrieval from the cold storage
+        documentModel = ColdStorageHelper.requestRetrievalFromColdStorage(session, documentModel.getRef(),
+                RESTORE_DURATION);
+        session.saveDocument(documentModel);
+        transactionalFeature.nextTransaction();
+        documentModel.refresh();
+
+        assertEquals(Boolean.TRUE,
+                documentModel.getPropertyValue(ColdStorageHelper.COLD_STORAGE_BEING_RETRIEVED_PROPERTY));
+
+        assertEquals(Boolean.TRUE, s3TestHelper.ifS3ObjectBeingRetrieved(documentModel));
+    }
+
+    @Override
+    protected void moveAndVerifyContent(CoreSession session, DocumentModel documentModel) throws IOException {
+        documentModel = moveContentToColdStorage(session, documentModel.getRef());
+        session.saveDocument(documentModel);
+        s3TestHelper.updateObjectClassStorageToGlacier(documentModel);
+        transactionalFeature.nextTransaction();
+        documentModel.refresh();
+
+        assertTrue(documentModel.hasFacet(ColdStorageHelper.COLD_STORAGE_FACET_NAME));
+
+        assertNull(documentModel.getPropertyValue(ColdStorageHelper.FILE_CONTENT_PROPERTY));
+
+        // check if the `coldstorage:coldContent` property contains the original file content
+        Blob content = (Blob) documentModel.getPropertyValue(ColdStorageHelper.COLD_STORAGE_CONTENT_PROPERTY);
+        assertNotNull(content);
+        assertEquals(FILE_CONTENT, content.getString());
+        assertEquals("glacier", ((ManagedBlob) content).getProviderId());
+    }
+
+    @Override
+    protected DocumentModel moveContentToColdStorage(CoreSession session, DocumentRef documentRef) {
+        DocumentModel documentModel = ColdStorageHelper.moveContentToColdStorage(session, documentRef);
+        session.saveDocument(documentModel);
+        // Mock AWS Lifecycle rule
```

Comment:
this part can be replaced by `super.moveContentToColdStorage(session, documentRef )` and add only the S3 part